component/main.jsonnet

```diff
@@ -1,10 +1,278 @@
+    'splunk-sourcetype': params.splunk.sourcetype,
+    'splunk-source': params.splunk.source,
+    'td-agent.conf': |||
+      <system>
+        log_level "#{ENV['LOG_LEVEL'] }"
+      </system>
+      <source>
+        @type  forward
+        @id    input1
+        port  24224
+        %(tls_config)s
+        <security>
+          shared_key "#{ENV['SHARED_KEY'] }"
+          self_hostname "#{ENV['HOSTNAME']}"
+        </security>
+      </source>
+      <match **>
+        @type splunk_hec
+        protocol "#{ENV['SPLUNK_PROTOCOL'] }"
+        insecure_ssl "#{ENV['SPLUNK_INSECURE'] }"
+        hec_host "#{ENV['SPLUNK_HOST'] }"
+        sourcetype "#{ENV['SPLUNK_SOURCETYPE'] }"
+        source "#{ENV['SPLUNK_SOURCE'] }"
+        index "#{ENV['SPLUNK_INDEX'] }"
+        hec_port "#{ENV['SPLUNK_PORT'] }"
+        hec_token "#{ENV['SPLUNK_TOKEN'] }"
+        host "#{ENV['NODE_NAME']}"
+        ssl_verify "#{ENV['SPLUNK_SSL_VERIFY']}"
+        ca_file /secrets/splunk/splunk-ca.crt
+        # TODO: configurize buffer config
+        <buffer>
+              @type memory
+              chunk_limit_records 100000
+              chunk_limit_size 200m
+              flush_interval 5s
+              flush_thread_count 1
+              overflow_action block
+              retry_max_times 3
+              total_limit_size 600m
+        </buffer>
+        # END: configurize buffer config
+      </match>
+    ||| % {
+      tls_config: if params.fluentd.ssl.enabled then |||
+        <transport tls>
+          cert_path /secret/fluentd/tls.crt
+          private_key_path /secret/fluentd/tls.key
+          private_key_passphrase "#{ENV['FLUENTD_SSL_PASSPHRASE'] }"
+        </transport>
+      |||,
+    },
+  },
+};
+
+local secret = kube.Secret(app_name) {
+  metadata+: {
+    labels+: {
+      'app.kubernetes.io/component': 'fluentd',
+    },
+  },
+  data: {
+    shared_key: std.base64(params.fluentd.sharedkey),
+    'hec-token': std.base64(params.splunk.token),
+    'fluentd-ssl-passsphrase': std.base64(params.fluentd.ssl.passphrase),
+    [if params.fluentd.ssl.enabled then 'forwarder-tls.crt']: std.base64(params.fluentd.ssl.cert),
+    [if params.fluentd.ssl.enabled then 'forwarder-tls.key']: std.base64(params.fluentd.ssl.key),
+    [if params.fluentd.ssl.enabled then 'ca-bundle.crt']: std.base64(params.fluentd.ssl.cert),
+  },
+};
+
+local secret_splunk = kube.Secret(app_name + '-splunk') {
+  metadata+: {
+    labels+: {
+      'app.kubernetes.io/component': 'fluentd',
+    },
+  },
+  data: {
+    'splunk-ca.crt': std.base64(params.splunk.ca),
+  },
+};
+
+local service_spec() = kube.Service(app_name) {
+  metadata+: {
+    labels+: {
+      'app.kubernetes.io/component': 'fluentd',
+    },
+  },
+  spec: {
+    ports: [ {
+      name: '24224-tcp',
+      protocol: 'TCP',
+      port: 24224,
+      targetPort: 24224,
+    } ],
+    selector: app_selector,
+    type: 'ClusterIP',
+    sessionAffinity: 'None',
+  },
+};
+local service = service_spec();
```

Comment:
seems unnecessary to have another local with identical value to `service_spec`.

Reply:
To be honest, i didn't realy understood why there are two services. I now only use the -headless service, as the forwarder is only used internally.